### PR TITLE
[misc] Remove useless Program::sync

### PR DESCRIPTION
**Pull request by PGZXB**
Issue: #

### Brief Summary
IIUC, the `Program::sync` is useless now.

### CODE REVIEW

File: taichi/program/program.cpp [hunk 1]
```diff
@@ -158,7 +158,6 @@ Program::Program(Arch desired_arch) : snode_rw_accessors_bank_(this) {
   SNode::counter = 0;
 
   result_buffer = nullptr;
-  sync = true;
```
Comment (bobcao3) [inline taichi/program/program.cpp:161]: what is this for originally?

**Review by bobcao3 — COMMENTED** [review]
LGTM

File: taichi/program/program.cpp (hunk 1 above)
Comment (PGZXB) [reply taichi/program/program.cpp:161]: For AsyncTaichi? The #5523 removed it. @lin-hitonami , WDYT?

**Review by ailzhang — APPROVED** [review]
Thanks!

File: taichi/program/program.cpp (hunk 1 above)
Comment (ailzhang) [reply taichi/program/program.cpp:161]: Yea IIRC this was legacy from async taichi.

**Pull request merged**